ocaml/xapi/extauth_plugin_ADpbis.ml

```diff
@@ -20,6 +20,35 @@ open D
 
 open Stdext.Xstringext
 
+let lwsmd_service = "lwsmd"
+
+let is_ad_enabled ~__context =
+  Helpers.get_localhost ~__context
+  |> (fun self -> Db.Host.get_external_auth_type ~__context ~self)
+  |> (fun x -> x = Xapi_globs.auth_type_AD_Likewise)
+
+let start_lwsmd ~__context =
+  (* if current host is auth with AD, then start lwsmd *)
+  try
+    ignore (Forkhelpers.execute_command_get_output !Xapi_globs.systemctl ["start"; lwsmd_service]);
+    Thread.delay !Xapi_globs.lwsmd_service_bootup_time;
+  with e -> error "Fail to start %s with error %s" lwsmd_service (ExnHelper.string_of_exn e)
+
+let init_lwsmd_service  ~__context =
+  (* This function is called during xapi start *)
+  (* it will start lwsmd service if the host is authed with AD, stop lwsmd service otherwise *)
+  let op = match (is_ad_enabled ~__context) with
+    | true -> "start"
+    | false -> "stop" in
```

Comment:
I'd like the service to be disabled by default instead of letting it start automatically and stopping it afterwards.

Reply:
yes, disable the auto-start of `lwsmd` is what I am trying to do . that is the first version.
But then I found the service is `Requred` and then bootup by `upgrade-likewise-to-pbis`. 
So the simplest way I found here is start/stop as it looks now.

Reply:
> So the simplest way I found here is start/stop as it looks now.

The problem with this is that now we have a race condition. If somehow systemd decides to start the AD service after xapi run stop the service will keep running.

Reply:
yes, that is the reason I change `xapi.service` to sync the bootup, you can see it in the top commit

Reply:
@psafont  per discussion with PM, this PR will not be backport to older XS which can be upgraded from XS 6.x, 
and XS 7.x all use `likewise` instead of `pbis`, 
so we can safely disable `upgrade-likewise-to-pbis` , I have updated the PR, please have a look.